src/hotspot/share/memory/universe.cpp

```diff
@@ -1255,23 +1239,27 @@ uintptr_t Universe::verify_mark_bits() {
 }
 #endif // PRODUCT
 
-
-void LatestMethodCache::init(Klass* k, Method* m) {
-  if (!UseSharedSpaces) {
-    _klass = k;
-  }
-#ifndef PRODUCT
-  else {
-    // sharing initialization should have already set up _klass
-    assert(_klass != nullptr, "just checking");
+void LatestMethodCache::init(JavaThread* current, InstanceKlass* ik,
+                             const char* method, Symbol* signature, bool is_static)
+{
+  TempNewSymbol name = SymbolTable::new_symbol(method);
+  Method* m = nullptr;
+  // The klass must be linked before looking up the method.
+  if (!ik->link_class_or_fail(current) ||
+      ((m = ik->find_method(name, signature)) == nullptr) ||
+      is_static != m->is_static()) {
+    ResourceMark rm(current);
+    // NoSuchMethodException doesn't actually work because it tries to run the
+    // <init> function before java_lang_Class is linked. Print error and exit.
+    vm_exit_during_initialization(err_msg("Unable to link/verify %s.%s method",
+                                 ik->name()->as_C_string(), method));
   }
-#endif
 
+  _klass = ik;
   _method_idnum = m->method_idnum();
   assert(_method_idnum >= 0, "sanity check");
 }
```

Comment:
Can you move this function up to where the accessor are too?   Like around line 930?

Reply:
Done